### PR TITLE
Restaking vaults

### PR DESCRIPTION
**Pull request by dfkadyr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -542,6 +542,52 @@ await sdk.vault.getValidators({
...
+#### Description:
+
+Returns eigen pods for restake vault.
+
+#### Arguments:
+
+| Name | Type | Required |
+|------|------|-------------|
+| vaultAddress | `string` | **Yes** | 
+| limit | `number` | **Yes** | To implement pagination |
```
Comment (mike-diamond) [inline README.md:556]: | limit | `number`         | **No** | Limits the number of eigen pods returned. Defaults to 100                                                                                     |
| skip | `number`          | **No** | Skips the specified number of eigen pods. Defaults to 0                                                                                       |

File: README.md [hunk 2]
```diff
@@ -1281,6 +1333,95 @@ const { data, to } = await sdk.vault.setDepositDataManager.encode(params)
...
+
+#### Description:
+
+Adding eigen pod to the vault. Only for restake vaults and only restake operators manager can perform this action.
+
+#### Arguments:
+
+| Name           | Type     | Required | Description |
+|----------------|----------|----------|-------------|
+| userAddress    | `string` | **Yes**  | - |
```
Comment (mike-diamond) [inline README.md:1346]: | userAddress           | `string` | **Yes**  | The address of the user making the request |
| vaultAddress          | `string` | **Yes**  | The address of the vault                   |

File: README.md [hunk 3]
```diff
@@ -1281,6 +1333,95 @@ const { data, to } = await sdk.vault.setDepositDataManager.encode(params)
...
+const { data, to } = await sdk.vault.createEigenPod.encode(params)
+// Get an approximate gas per transaction
+const gas = await sdk.vault.createEigenPod.estimateGas(params)
+```
+---
+### `sdk.vault.setEigenPodOperator`
+
+#### Description:
+
+Adding operator to the current eigen pod. Only for restake vaults and only restake operators manager can perform this action.
```
Comment (mike-diamond) [inline README.md:1369]: Mb something like that?
This action is specific to restake vaults and can only be executed by the restake operators manager.

File: src/methods/vault/transactions/setEigenPodOperator/index.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const signer = await provider.getSigner(userAddress)
+  const signedEigenPodOwnerContract = eigenPodOwnerContract.connect(signer)
+
+  const response = await signedEigenPodOwnerContract.delegateTo(
+    operatorAddress,
+    {
+      signature: '0x',
+      expiry: 0,
+    },
+    '0x0000000000000000000000000000000000000000000000000000000000000000'
```
Comment (mike-diamond) [inline src/methods/vault/transactions/setEigenPodOperator/index.ts:22]: ZeroAddress?

**Review by mike-diamond — COMMENTED** [review]
💯

File: README.md (hunk 1 above)
Comment (dfkadyr) [reply README.md]: [v3]: improves, add RestakingVaultAbi, edit README

File: README.md (hunk 2 above)
Comment (dfkadyr) [reply README.md]: https://github.com/stakewise/v3-sdk/pull/122/commits/210ca5e6df93a0afb07491dae920bdce608a2a9b#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R1346

File: README.md (hunk 1 above)
Comment (dfkadyr) [reply README.md]: https://github.com/stakewise/v3-sdk/pull/122/commits/210ca5e6df93a0afb07491dae920bdce608a2a9b#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R510

File: README.md (hunk 3 above)
Comment (dfkadyr) [reply README.md]: https://github.com/stakewise/v3-sdk/pull/122/commits/210ca5e6df93a0afb07491dae920bdce608a2a9b#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R1369

File: src/methods/vault/transactions/setEigenPodOperator/index.ts (hunk 4 above)
Comment (dfkadyr) [reply src/methods/vault/transactions/setEigenPodOperator/index.ts:22]: yep, its "approverSalt", we use static value

File: src/contracts/createContracts.ts [hunk 5]
```diff
@@ -133,6 +134,7 @@ export const createContracts = (input: CreateContractsInput) => {
       createVault: (address: string) => createContract<StakeWise.ABI.Vault>(address, VaultAbi, provider),
       createUniswapPool: (address: string) => createContract<StakeWise.ABI.UniswapPool>(address, UniswapPoolAbi, provider),
       createPrivateVault: (address: string) => createContract<StakeWise.ABI.PrivateVault>(address, PrivateVaultAbi, provider),
+      createEigenPodOwner: (address: string) => createContract<StakeWise.ABI.EigenPodOwner>(address, EigenPodOwnerAbi, provider),
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/contracts/createContracts.ts:138]: https://github.com/stakewise/v3-sdk/pull/122/commits/210ca5e6df93a0afb07491dae920bdce608a2a9b#diff-fe42d8cad841f79a8eef9e2fce831f7d71735446ea7e110867b85d706c89b573R139

File: src/contracts/abis/VaultAbi.json [hunk 6]
```diff
@@ -536,5 +536,51 @@
...
+    ],
+    "stateMutability": "view",
+    "type": "function"
+  },
+  {
+    "inputs": [],
+    "name": "createEigenPod",
+    "outputs": [],
+    "stateMutability": "nonpayable",
+    "type": "function"
```
[the remark this answers is not in the document]
Comment (dfkadyr) [reply src/contracts/abis/VaultAbi.json]: https://github.com/stakewise/v3-sdk/pull/122/commits/210ca5e6df93a0afb07491dae920bdce608a2a9b#diff-c1b67827e802b529eaa9e9a5aa248f87c21c441ac0bff3206ea03f708dcfaf93R1

**Review by mike-diamond — APPROVED** [review]

**Pull request merged**